tokio/src/runtime/time/mod.rs

```diff
@@ -337,7 +351,7 @@ impl Handle {
 
                     waker_list.wake_all();
 
-                    lock = self.inner.lock_sharded_wheel(id);
+                    lock = wheels_lock.lock_sharded_wheel(id);
```

Comment:
You need to unlock `wheels_lock` here too. It's important that we don't hold any locks when calling `wake_all`.

Reply:
Oh yeah I forgot that! Added it in [68e5cf1](https://github.com/tokio-rs/tokio/pull/6779/commits/68e5cf179f902d62b3d9605bd48de3433390548b).